gmso/tests/test_expression.py

```diff
@@ -0,0 +1,243 @@
+        expression.expression = 'a^(2*x) + b^(2*y) + c^(2*z)'
+        assert sympy.Symbol('x') in expression.independent_variables
+
+    def test_set_expression_invalid(self):
+        expression = _PotentialExpression(
+            'a^x + b^y + c^z',
+            independent_variables={'x', 'y', 'z'},
+            parameters={
+                'a': 2.6 * u.nm,
+                'b': 2.7 * u.nm,
+                'c': 22.8 * u.hertz
+            }
+        )
+        with pytest.raises(ValueError) as e:
+            expression.expression = '2 * theta^2 + 3 * phi^2'
+
+        assert sympy.sympify('a^x + b^y + c^z') == expression.expression
+
+    def test_set_parameters(self):
+        expression = _PotentialExpression(
+            'a^x + b^y + c^z',
+            independent_variables={'x', 'y', 'z'},
+            parameters={
+                'a': 2.6 * u.nm,
+                'b': 2.7 * u.nm,
+                'c': 22.8 * u.hertz
+            }
+        )
+        expression.parameters = {
+            'a': 2.7 * u.nm,
+            'b': 2.8 * u.nm,
+            'c': 220.0 * u.hertz
+        }
+        assert expression.parameters['a'] == u.unyt_quantity(2.7, units='nm')
+        assert expression.parameters['b'] == u.unyt_quantity(2.8, units='nm')
+        assert expression.parameters['c'] == u.unyt_quantity(220.0, units='hertz')
+
+    def test_set_parameters_extra(self):
+        expression = _PotentialExpression(
+            'a^x + b^y + c^z',
+            independent_variables={'x', 'y', 'z'},
+            parameters={
+                'a': 2.6 * u.nm,
+                'b': 2.7 * u.nm,
+                'c': 22.8 * u.hertz
+            }
+        )
+
+        expression.parameters = {
+            'a': 2.7 * u.nm,
+            'b': 2.8 * u.nm,
+            'c': 220.0 * u.hertz,
+            'd': 229.0 * u.hertz
+        }
+
+        assert expression.parameters['a'] == u.unyt_quantity(2.7, units='nm')
+        assert expression.parameters['b'] == u.unyt_quantity(2.8, units='nm')
+        assert expression.parameters['c'] == u.unyt_quantity(220.0, units='hertz')
+        assert 'd' not in expression.parameters
+
+    def test_set_parameters_invalid(self):
+        expression = _PotentialExpression(
+            'a^x + b^y + c^z',
+            independent_variables={'x', 'y', 'z'},
+            parameters={
+                'a': 2.6 * u.nm,
+                'b': 2.7 * u.nm,
+                'c': 22.8 * u.hertz
+            }
+        )
+        with pytest.raises(ValueError):
+            expression.parameters = {
+                    'l': 2.7 * u.nm,
+                    'm': 2.8 * u.nm,
+                    'n': 220.0 * u.hertz
+                }
+
+        assert expression.parameters['a'] == u.unyt_quantity(2.6, units='nm')
+        assert expression.parameters['b'] == u.unyt_quantity(2.7, units='nm')
+        assert expression.parameters['c'] == u.unyt_quantity(22.8, units='hertz')
+        assert 'w' not in expression.parameters
```

Comment:
Could we instead use one of the values that raised the value error above. i dont think those are in scope, but would be a small change.
```suggestion
        assert 'm' not in expression.parameters
```